query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryExecutor.scala

```diff
@@ -447,9 +447,12 @@ class QueryExecutor(httpClient: HttpClient,
                           explain: Boolean): Result = {
     val serializedAnalyses = serializeAnalysis(analyses)
 
-    val wantQueryTimeoutSeconds = queryTimeoutSeconds.map(x => try { Integer.parseInt(x) } catch { case _: NumberFormatException => queryTimeoutSecondsMax })
-                                                     .getOrElse(queryTimeoutSecondsMax)
-    val qtos = math.min(queryTimeoutSecondsMax, wantQueryTimeoutSeconds)
+    val qtos = queryTimeoutSeconds.map(x => try { Integer.parseInt(x) } catch { case _: NumberFormatException => defaultQueryTimeoutSeconds })
+                                  .getOrElse(defaultQueryTimeoutSeconds)
+
+    if (qtos > defaultQueryTimeoutSeconds && !debug) {
+      return RequestTimeoutExceedLimit(defaultQueryTimeoutSeconds)
```

Comment:
it looks like previously if you sent a really high timeout, it would just get swapped out in favor of the "max" setting. Now we'll return an error-- which is more explicit feedback.

Have you checked if anyone is sending high query timeouts right now (and will suddenly start hitting this error)?

Reply:
The http timeout maxed out at 10 mins.  We only changed sql timeout.  Nothing asking for higher than 10 mins should have worked.
I think collimator was the only client trying to ask for more than 10 mins.

Reply:
Looks like collimator is using 15mins.  I don't think it gets what it asks for though.

Reply:
I might be misunderstanding the code, maybe you can help me figure out where.

It looks like this old line:
```
val qtos = math.min(queryTimeoutSecondsMax, wantQueryTimeoutSeconds)
```
would silently override any requested timeout that was passed in with the max possible timeout, but still allow the query to run.

But the new code will throw an exception if the requested timeout is too high.

Reply:
The problem is in QueryResource line 203
base = base.receiveTimeoutMS(receiveTimeout.toMillis.toInt).connectTimeoutMS(connectTimeout.toMillis.toInt)

receiveTimeout is limited to 10mins.

Reply:
I can see on that line how before, we were previously always setting the `receiveTimeoutMS` of the requestBuilder using receiveTimeout from config, and now with your changes we will use queryTimeoutSeconds (from http parameters) if possible, and only call back to receiveTimeout.

My question is about this: It looks like previously, API users could send really really big `queryTimeoutSeconds` as a parameter, and the worst that would happen is that it would get overriden the max setting, but their request would still be allowed to run (and possibly complete in time). But now it looks like if they send a request with `queryTimeoutSeconds` set larger than `defaultQueryTimeoutSeconds` (for a non debug request), they will get an immediate error.

Have we looked to see if there will be many users impacted by this?